### PR TITLE
Issue/303/mockdata tests

### PR DESCRIPTION
**Pull request by rherbonnet**
Added tests for the new mockdata module. Checks mass estimation is correct for 8 different masses and initial guesses, verifies the chang redshift distribution, and the addition of Gaussian shape noise.

### CODE REVIEW

**Comment by combet** [comment]
@rherbonnet, several PR have been merged to master during the week, following our discussion last Friday. As mentionned, the API of some functions has slightly change and this need to be propagated to your `test_mock_data.py` test suite.

- `Delta` has become an optional keyword (default=200) in `mock_data` and should not be passed as an argument anymore, i.e. `data = mock.generate_galaxy_catalog(10**mass, 0.3, 4, cosmo, 200, 0.8, ngals=ngals)` should be changed to `data = mock.generate_galaxy_catalog(10**mass, 0.3, 4, cosmo, 0.8, ngals=ngals)`

- `make_binned_profile` has been renamed `make_radial_profile`

- `polaraveraging.py` was renamed `dataops.py`, so the import should be changed

- With the new structure of `dataops.py`, the syntax `pa.compute_tangential_and_cross_components(cl, geometry="flat")` is not valid anymore. When working with a GalaxyCluster object, use `cl.compute_tangential_and_cross_components(geometry="flat")` instead.

- Similarly, `binned = pa.make_binned_profile(cl, "radians", "Mpc", bins=pa.make_bins(0.5, 5.0, 100), cosmo=cosmo, include_empty_bins=False)` should be updated to `binned = cl.make_radial_profile("Mpc", bins=pa.make_bins(0.5, 5.0, 100), cosmo=cosmo, include_empty_bins=False)`

File: tests/test_mockdata.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    
+    
+    data = mock.generate_galaxy_catalog(10**mass, 0.3, 4, cosmo, 200, 'chang13', ngals=ngals,
+                                        zsrc_min=zmin, zsrc_max=zmax)
+    # Check that there all galaxies are within the given limits
+    assert_equal(np.count_nonzero((data['z']<zmin)|(data['z']>zmax)),0)
+    # Check that the z distribution follows Chang13 distribution
+    hist = np.histogram(data['z'],bins=bins)
+    chang = np.array([mock._chang_z_distrib(z) for z in bins[:-1]+0.05])
+    assert_allclose(hist[0],chang*ngals/2,atol=100,rtol=0.1)
```
Comment (shenmingfu) [inline tests/test_mockdata.py:99]: Thank you very much for the code. Do you mind if I ask a question about this line? (line 99)
Does this 1./2. factor come from these steps? 
```
integrate.quad(_chang_z_distrib, zmin, zmax) ~ 0.2 => normalization factor
bin_size = 0.1
=> 1/0.2*0.1 = 1./2.
```
Thank you.

**Review by shenmingfu — COMMENTED** [review]
Thank you for the code. Could you take a look at my comments above? Thanks.

File: tests/test_mockdata.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    np.random.seed(285713)
+
+    
+    data = mock.generate_galaxy_catalog(10**15., 0.3, 4, cosmo, 0.8, ngals=50000,shapenoise=0.5)
+    # Check that there are no galaxies with |e|>1
+    assert_equal(np.count_nonzero((data['e1']>1) | (data['e1']<-1)),0)
+    assert_equal(np.count_nonzero((data['e2']>1) | (data['e2']<-1)),0)
+    
+    
+    data = mock.generate_galaxy_catalog(10**12., 0.3, 4, cosmo, 0.8, ngals=50000,shapenoise=0.25)
```
[the remark this answers is not in the document]
Comment (rherbonnet) [reply tests/test_mockdata.py:118]: yes, good point! Now the docstring is incorrect and just a copy paste. Good catch

File: tests/test_mockdata.py [hunk 3]
```diff
@@ -65,7 +65,7 @@ def mass_mock_cluster(mass=15.,guess=15.):
     
 def test_z_distr():
     """
-    Test the redshift distribution
+    Test the redshift distribution options: single plan, uniform, Chang13.
```
Comment (shenmingfu) [inline tests/test_mockdata.py:68]: Perhaps a typo? plan -> plane

**Review by shenmingfu — APPROVED** [review]
Looks good. Thanks.

**Pull request merged**